Doc/library/typing.rst

```diff
@@ -588,9 +588,9 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
 .. data:: Union
 
-   Union type; ``Union[X, Y]`` means either X or Y.
+   Union type; ``Union[X, Y]`` is equivalent to ``X | Y`` and means either X or Y.
 
-   To define a union, use e.g. ``Union[int, str]``.  Details:
+   To define a union, use e.g. ``Union[int, str]`` or the shorthand ``int | str``.  Details:
 
    * The arguments must be types and there must be at least one.
```

Comment:
Out of curiosity, why did you change the ``redundant arguments are skipped`` bullet point, but not the others? I can see that ``Unions of a single argument vanish`` may not apply directly to the new syntax, but the others do right?

If we manage to merge them, then perhaps we can remove the ``union_object == other`` section here https://docs.python.org/3.10/library/stdtypes.html#types-union and just link to the typing version. That will save us from some repetition.

Reply:
I went a bit back and forth over this. In the end I decided that most of them apply mostly to `Union`:

* Unions of unions don't really exist for PEP 604 unions, except when counting `(X | Y) | Z`. So we could write one of the following:
    + `Union[Union[int, str], float] == Union[int, str, float] == int | str | float`
    + `Union[Union[int, str], float] == int | str | float`
  I kind of like the second one, since it emphasized the new syntax, but I though it might be a bit confusing.
* The redundant arguments should probably be changed to either list both `Union[int, str, int] == Union[int, str]` and `int | str | int == `int | str` or only the latter (my preference). No idea why I chose this strange middle thing.
* Same of comparing unions, really.

Reply:
You points make sense. I would like to not outright replace the examples under ``typing.Union`` with only the PEP 604 version. For the same reason that we haven't changed examples under `typing.List` [1] to PEP 585 style `list` and instead just changed every other use of `typing.List` (same for every other PEP 585-ied type really). Considering at runtime they aren't the same, I'd hate if beginners tripped over this thinking they were.

This is one tough pickle :(. So I'm OK with the status quo.

[1] https://docs.python.org/3.11/library/typing.html#typing.List